src/rust/util/src/bb02_async.rs

```diff
@@ -74,6 +74,38 @@ pub fn yield_now() -> impl core::future::Future<Output = ()> {
     })
 }
 
+/// Executes both futures in tandem, returning the results of both when both are done.
```

Comment:
This is fine, but I want to point out that there are libraries already implementing it:

* https://docs.rs/futures-lite/latest/futures_lite/future/fn.zip.html
* https://docs.rs/futures/latest/futures/macro.join.html
* [nightly only] https://doc.rust-lang.org/std/future/macro.join.html